skymp5-client/src/features/loadOrder.ts

```diff
@@ -6,151 +6,151 @@ import { getScreenResolution } from "../view/formView";
-      if (sp.settings['skymp5-client']['ignoreLoadOrderMismatch']) {
-        updateText(
-          'LOAD ORDER ERROR!\nHowever, ignoring it because of ignoreLoadOrderMismatch being set.' +
-          '\nExpect EVERYTHING BREAK, unless you know what you are doing.\nCheck console for details.' +
-          '\nThis message will disappear after 30 seconds.',
-          [255, 0, 0, 1], 30,
-        );
-        return;
-      }
-      updateText(
-        'LOAD ORDER ERROR!\nCheck console for details.',
-        [255, 0, 0, 1],
-      );
-      sp.browser.loadUrl('about:blank');
-    });
+    resetText();
+    const clientMods = getClientMods();
+    printModOrder('Client load order:', clientMods);
+    return getServerMods(5)
+        .then((serverMods) => {
+            printModOrder('Server load order:', serverMods);
+            if (clientMods.length < serverMods.length) {
+                throw new Error(`Missing some server mods. Server has ${serverMods.length}, we have ${clientMods.length}`);
+            }
+            if (clientMods.length > serverMods.length) {
+                updateText(
+                    'LOAD ORDER WARNING: you have more mods than server!\n(or could not receive server mod list)\nCheck console for details.',
+                    [255, 255, 0, 1], 5,
+                );
+            }
+            let fail = [];
+            for (let i = 0; i < serverMods.length; ++i) {
+                // Need case-insensitive check for 1.6+
+                if (
+                    clientMods[i].filename.toLowerCase() !== serverMods[i].filename.toLowerCase() ||
+                    clientMods[i].size !== serverMods[i].size ||
+                    clientMods[i].crc32 !== serverMods[i].crc32
+                ) {
+                    fail.push(i);
+                    printConsole(`${i}-th mod (numbered from 0) does not match.`);
+                    printConsole(`Server has ${JSON.stringify(serverMods[i])}`);
+                    printConsole(`We have ${JSON.stringify(clientMods[i])}`);
+                }
+            }
+            if (fail.length !== 0) {
+                throw new Error('Load order check failed! Indices: ' + JSON.stringify(fail));
+            }
+        })
+        .catch((err) => {
+            printConsole(err);
+            if (sp.settings['skymp5-client']['ignoreLoadOrderMismatch']) {
+                updateText(
+                    'LOAD ORDER ERROR!\nHowever, ignoring it because of ignoreLoadOrderMismatch being set.' +
+                    '\nExpect EVERYTHING BREAK, unless you know what you are doing.\nCheck console for details.' +
+                    '\nThis message will disappear after 30 seconds.',
+                    [255, 0, 0, 1], 30,
+                );
+                return;
+            }
+            updateText(
+                'LOAD ORDER ERROR!\nCheck console for details.',
+                [255, 0, 0, 1]
+            );
+            sp.browser.loadUrl('about:blank');
+        });
 };
```

Comment:
Do we need this indentation change? It's hard to check actual changes.

Reply:
> Do we need this indentation change? It's hard to check actual changes.

fixed